File: src/main/java/cz/stasimek/fakturaceeasypeasy/enumeration/VatType.java
package cz.stasimek.fakturaceeasypeasy.enumeration;

import java.util.Arrays;
import java.util.LinkedHashMap;
import java.util.Map;
import java.util.stream.Collectors;
import lombok.Getter;

@Getter
public enum VatType {
	NOT_VAT_PAYER("Neplátce DPH"),
	VAT_PAYER("Plátce DPH"),
	IDENTIFIED_PERSON("Identifikovaná osoba");

	private final String nameCs;

	VatType(String nameCs) {
		this.nameCs = nameCs;
	}

	public static Map<String, String> valuesMap() {
		return Arrays.stream(VatType.values())
				.map((VatType v) -> new String[]{v.name(), v.getNameCs()})
				.collect(Collectors.toMap(p -> p[0], p -> p[1], (x, y) -> y, LinkedHashMap::new));
	}
}

File: src/main/kotlin/cz/stasimek/fakturaceeasypeasy/enumeration/VatType.kt
package cz.stasimek.fakturaceeasypeasy.enumeration

enum class VatType(private val nameCs: String) {
    NOT_VAT_PAYER("Neplátce DPH"),
    VAT_PAYER("Plátce DPH"),
    IDENTIFIED_PERSON("Identifikovaná osoba");

    companion object {
        fun valuesMap(): Map<String, String> {
            return values()
                .map { it.name to it.nameCs }
                .toMap(LinkedHashMap())
        }
    }
}